### PR TITLE
PythonPackage: url -> pypi

### PR DESCRIPTION
**Pull request by adamjstewart**
This is a continuation of #17587

### CODE REVIEW

File: var/spack/repos/builtin/packages/awscli/package.py [hunk 1]
```diff
@@ -11,8 +11,7 @@ class Awscli(PythonPackage):
     """This package provides a unified command line interface to
        Amazon Web Services"""
 
-    homepage = "https://pypi.org/project/awscli/"
-    url      = "https://pypi.io/packages/source/a/awscli/awscli-1.16.308.tar.gz"
+    pypi = "awscli/awscli-1.16.308.tar.gz"
```
Comment (cosmicexplorer) [inline var/spack/repos/builtin/packages/awscli/package.py:14]: Was vaguely iffy about removing the `homepage`, but I think this result looks super readable.

**Review by cosmicexplorer — APPROVED** [review]
Let me know if there's anything specific you'd like another look over! Otherwise I really like this.

File: var/spack/repos/builtin/packages/awscli/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/awscli/package.py:14]: Yeah, as far as `spack info` is concerned, nothing has changed. It's just defined in the base class now.

**Review by tgamblin — APPROVED** [review]
This looks great to me. I was going to ask if we could omit the per-package pypi directory and just put the archive name, but I answered my own question and apparently we cannot:

```console
(spackle):spack> git diff develop | grep 'pypi =' | grep -o '"[^\"]*"'| sed 's/"//g' | pcregrep -v '([^/]+)/\1'
advancedhtmlparser/AdvancedHTMLParser-8.1.4.tar.gz
aiohttp_cors/aiohttp-cors-0.7.0.tar.gz
catkin-pkg/catkin_pkg-0.4.23.tar.gz
cython/Cython-0.29.21.tar.gz
dendropy/DendroPy-4.3.0.tar.gz
gitpython/GitPython-3.1.7.tar.gz
gpy/GPy-1.9.9.tar.gz
gsi/GSI-0.6.5.tar.gz
jupyter-client/jupyter_client-6.1.7.tar.gz
jupyter-core/jupyter_core-4.6.0.tar.gz
jupyterlab-pygments/jupyterlab_pygments-0.1.1.tar.gz
Keras-Preprocessing/Keras_Preprocessing-1.1.2.tar.gz
markdown/Markdown-2.6.11.tar.gz
mypy-extensions/mypy_extensions-0.4.3.tar.gz
nest-asyncio/nest_asyncio-1.4.0.tar.gz
Onnx/onnx-1.6.0.tar.gz
pint/Pint-0.11.tar.gz
py6s/Py6S-1.8.0.tar.gz
Pybtex/pybtex-0.21.tar.gz
pygments-pytest/pygments_pytest-1.2.0.tar.gz
pygps/pyGPs-1.3.5.tar.gz
pymysql/PyMySQL-0.9.2.tar.gz
pypeg2/pyPEG2-2.15.2.tar.gz
pytest-check-links/pytest_check_links-0.3.4.tar.gz
queryablelist/QueryableList-3.1.0.tar.gz
sqlalchemy-utils/SQLAlchemy-Utils-0.36.8.tar.gz
tensorboardx/tensorboardX-1.8.tar.gz
tpot/TPOT-0.11.5.tar.gz
treehash/TreeHash-1.0.2.tar.gz
unidecode/Unidecode-1.1.1.tar.gz
```

Kind of a bummer that they don't regularize the case and `_`'s in the tarball names, but oh well.  Thanks for all this refactoring!

**Pull request merged**